File: frontend/src/services/paymentService.js
import api from "../api/api";

// =====================================================
// GET ALL PAYMENTS
// =====================================================

export const getPayments = () => {

    return api.get("/payments");

};


// =====================================================
// GET PAYMENT BY ID
// =====================================================

export const getPaymentById = (id) => {

    return api.get(`/payments/${id}`);

};


// =====================================================
// CREATE PAYMENT
// =====================================================

export const createPayment = (payment) => {

    return api.post("/payments", payment);

};


// =====================================================
// UPDATE PAYMENT
// =====================================================

export const updatePayment = (id, payment) => {

    return api.put(`/payments/${id}`, payment);

};


// =====================================================
// DELETE PAYMENT
// =====================================================

export const deletePayment = (id) => {

    return api.delete(`/payments/${id}`);

};


// =====================================================
// GET PAYMENTS BY RESERVATION
// =====================================================

export const getPaymentsByReservation = (reservationId) => {

    return api.get(
        `/payments/reservation/${reservationId}`
    );

};


// =====================================================
// GET PAYMENTS BY STATUS
// =====================================================

export const getPaymentsByStatus = (status) => {

    return api.get(
        `/payments/status/${status}`
    );

};


// =====================================================
// GET PAYMENTS BY METHOD
// =====================================================

export const getPaymentsByMethod = (method) => {

    return api.get(
        `/payments/method/${method}`
    );

};

File: backend/src/main/java/com/hotelmanagement/service/PaymentService.java
package com.hotelmanagement.service;

import java.util.List;

import com.hotelmanagement.dto.request.PaymentRequest;
import com.hotelmanagement.dto.response.PaymentResponse;
import com.hotelmanagement.enums.PaymentMethod;
import com.hotelmanagement.enums.PaymentStatus;

public interface PaymentService {

    // ==========================================
    // CREATE
    // ==========================================

    PaymentResponse createPayment(PaymentRequest request);

    // ==========================================
    // UPDATE
    // ==========================================

    PaymentResponse updatePayment(
            Integer paymentId,
            PaymentRequest request);

    // ==========================================
    // DELETE
    // ==========================================

    void deletePayment(Integer paymentId);

    // ==========================================
    // GET BY ID
    // ==========================================

    PaymentResponse getPaymentById(Integer paymentId);

    // ==========================================
    // GET ALL
    // ==========================================

    List<PaymentResponse> getAllPayments();

    // ==========================================
    // SEARCH
    // ==========================================

    List<PaymentResponse> findByReservation(Integer reservationId);

    List<PaymentResponse> findByPaymentStatus(
            PaymentStatus paymentStatus);

    List<PaymentResponse> findByPaymentMethod(
            PaymentMethod paymentMethod);

}
